CMakeLists.txt

```diff
@@ -199,13 +199,13 @@ endif()
 ####
 # Setup the compiler options
 
-# set c++ standard to c++11.
+# set c++ standard to c++14.
 # Note: not working on CMake 2.8. We assume that user has
-#       a compiler with C++11 support.
+#       a compiler with C++14 support.
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-message(STATUS "C++11 support has been enabled by default.")
+message(STATUS "C++14 support has been enabled by default.")
```

Comment:
> everybody okay to require c++14?

That should be decided by nyanp.

If we'll use C++14, `/README.md` file should be updated. (But this can be done in future PR)

> This doesn't require full c++14 coverage but just enable c++14 flag right? Probably we need to require minimal clang/msvc/gcc version

That's a good idea. 

And checking compiler features with ...
http://en.cppreference.com/w/cpp/compiler_support
http://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros
might help us to be ready for perceived issues from the users that try to use the library on outdated platforms.